lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,345 @@
+import 'package:flutter/material.dart';
+import '../api/model/model.dart';
+import '../generated/l10n/zulip_localizations.dart';
+import '../model/narrow.dart';
+import '../model/store.dart';
+import 'color.dart';
+import 'content.dart';
+import 'icons.dart';
+import 'message_list.dart';
+import 'store.dart';
+import 'text.dart';
+import 'theme.dart';
+
+void showNewDmSheet(BuildContext context) {
+  final store = PerAccountStoreWidget.of(context);
+  showModalBottomSheet<void>(
+    context: context,
+    clipBehavior: Clip.antiAlias,
+    useSafeArea: true,
+    isScrollControlled: true,
+    builder: (BuildContext context) => Padding(
+      // By default, when software keyboard is opened, the ListView
+      // expands behind the software keyboard — resulting in some
+      // list entries being covered by the keyboard. Add explicit
+      // bottom padding the size of the keyboard, which fixes this.
+      padding: EdgeInsets.only(bottom: MediaQuery.viewInsetsOf(context).bottom),
+      child: SafeArea(
+        child: PerAccountStoreWidget(
+          accountId: store.accountId,
+          child: NewDmPicker(pageContext: context, store: store)))));
+}
+
+@visibleForTesting
+class NewDmPicker extends StatefulWidget {
+  const NewDmPicker({
+    super.key,
+    required this.pageContext,
+    required this.store,
+  });
+
+  final BuildContext pageContext;
+  final PerAccountStore store;
```

Comment:
I think we can skip passing this by replacing the `_updateFilteredUsers` call with an overridden `didChangeDependencies`.

```
  @override
  void didChangeDependencies() {
    super.didChangeDependencies();
    _handleSearchUpdate();
  }
```